packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -79,27 +81,34 @@ export class EditorPreviewManager extends WidgetOpenHandler<EditorPreviewWidget|
             } else {
                 this.shell.addWidget(editorWidget, {area: 'main'});
             }
-            preview.close();
+            preview.dispose();
             this.shell.activateWidget(editorWidget.id);
-            this.currentEditorPreview = undefined;
+            this.currentEditorPreview = Promise.resolve(undefined);
         });
     }
 
-    protected isCurrentPreviewUri(uri: URI): boolean {
-        const currentUri = this.currentEditorPreview && this.currentEditorPreview.getResourceUri();
+    protected async isCurrentPreviewUri(uri: URI): Promise<boolean> {
+        const editorPreview = await this.currentEditorPreview;
```

Comment:
can be `undefined`? `canHandle` is a caller.

Reply:
Yes, this can be undefined. The resulting call is equivalent to ``` const editorPreview = await undefined``` which evaluates to undefined (should not produce an error).